### PR TITLE
fix #1430: resolve namespace OIDC tenants dynamically to prevent OidcProxy NPE

### LINKED ISSUE #1430: Router crash-loop: OidcProxy NPE when OIDC server is slow to respond
## Description

The router backend crashes during startup with a `NullPointerException` in `OidcProxy.<init>` when the OIDC server (Keycloak) is reachable but takes longer than 10 seconds per tenant to respond to discovery requests.

## Steps to Reproduce

1. Deploy Keycloak on OpenShift with the `wanaku` realm
2. Deploy a `WanakuRouter` CR with `authServer: "http://keycloak:8080"`
3. The router pod starts and sequentially attempts OIDC discovery for 12 tenants (ns-0..ns-9, mcp, Default), each timing out after 10s
4. After all tenants timeout (~120s), the `OidcProxy` constructor crashes

## Error

```
java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.NullPointerException: Cannot invoke "io.quarkus.oidc.runtime.OidcProviderClientImpl.getWebClient()" because the return value of "io.quarkus.oidc.runtime.TenantConfigContext.getOidcProviderClient()" is null
	at io.quarkus.oidc.proxy.runtime.OidcProxy.<init>(OidcProxy.java:71)
	at io.quarkus.oidc.proxy.runtime.OidcProxyRecorder.setupRoutes(OidcProxyRecorder.java:20)
```

## Environment

- OpenShift 4.x (IBM Cloud)
- Image: `quay.io/orpiske/wanaku-router-backend:ci-refresh-helm-files`
- Keycloak: `quay.io/keycloak/keycloak:26.6.1`

## Notes

- Keycloak IS reachable from inside the cluster (`curl http://keycloak:8080/realms/wanaku` works from the operator pod)
- The OIDC tenants log "OIDC server is not available after a 10 seconds timeout" as warnings, but the `OidcProxy` does not handle the null provider client gracefully
- This blocks all authenticated router deployments on OpenShift

### PR DESCRIPTION
**Pull request by orpiske**
## Summary

- Replace 10 static `quarkus.oidc.ns-*` tenant definitions with a `NamespaceTenantConfigResolver` that builds tenant config dynamically on first request, eliminating sequential OIDC discovery at startup
- Add `connection-delay=60S` for the remaining static tenants (Default, `mcp`) as defense-in-depth against slow Keycloak responses
- Remove ns-* tenant disabling from `AuthConfigSource` (no longer needed since dynamic tenants are not initialized at startup when OIDC is disabled)

## Test plan

- [x] Unit tests for `NamespaceTenantConfigResolver` (14 tests): valid paths, boundary conditions, config values
- [x] Updated `AuthConfigSourceTest` (12 tests): removed ns-* assertions
- [x] All 349 unit tests pass
- [x] OIDC integration tests pass
- [ ] Deploy on OpenShift with Keycloak and verify router starts without NPE
- [ ] Verify namespace MCP endpoints return correct 401 challenges with resource metadata

## Summary by Sourcery

Dynamically resolve per-namespace OIDC tenants at request time to avoid startup discovery issues and OidcProxy NPE when the OIDC server is slow.

Enhancements:
- Introduce NamespaceTenantConfigResolver to build namespace-specific OIDC tenant configurations on demand instead of via static properties.
- Add OIDC connection delay configuration for the default and mcp tenants to tolerate slow Keycloak responses.

Tests:
- Add unit tests covering NamespaceTenantConfigResolver behavior for valid, invalid, and boundary namespace paths.
- Update AuthConfigSource tests to remove expectations around disabled per-namespace OIDC tenants when auth is set to none.

### CODE REVIEW

File: apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/support/NamespaceTenantConfigResolver.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        } catch (NumberFormatException e) {
+            return Uni.createFrom().nullItem();
+        }
+
+        if (nsIndex < 0 || nsIndex >= MAX_NAMESPACES) {
+            return Uni.createFrom().nullItem();
+        }
+
+        String tenantId = "ns-" + nsIndex;
+        String authServerUrl = authServer + "/realms/" + authRealm;
```
[the remark this answers is not in the document]
Comment (orpiske) [reply apps/wanaku-router-backend/src/main/java/ai/wanaku/backend/support/NamespaceTenantConfigResolver.java]: Good catch — applied. Added `stripTrailingSlash()` for both `authServer` and `authProxy` before URL construction, plus a test for the trailing-slash case. See 7eddc8e.
